test/changefeed.test.js

```diff
@@ -33,183 +34,292 @@ if (CONFIG.experimental_cloudapi_changefeed !== true) {
     process.exit(0);
 }
 
+function makeHttpClient(client) {
+    const httpClient = restify.createHttpClient({
+        url: client.url.href,
+        version: client.headers['accept-version'],
+        retryOptions: {
+            retry: 0
+        },
+        log: client.log.child({
+            component: 'changefeed',
+            level: 'trace'
+        }),
+        rejectUnauthorized: false,
+        signRequest: client.signRequest
+    });
+    httpClient.keyId = client.keyId;
+    httpClient.privateKey = client.privateKey;
+    httpClient.publicKey = client.publicKey;
+    return httpClient;
+}
+
 test('Changefeed test', function (suite) {
-    common.setup({
-        clientApiVersion: '~9'
-    }, function (_, clients, server) {
-        const CLIENTS = clients;
-        const CLIENT = clients.user;
-        const OTHER = clients.other;
-        const SERVER = server;
-        const shed = new Watershed();
-
-
-        const httpClient = restify.createHttpClient({
-            url: CLIENT.url.href,
-            version: CLIENT.headers['accept-version'],
-            retryOptions: {
-                retry: 0
-            },
-            log: CLIENT.log.child({
-                component: 'changefeed',
-                level: 'trace'
-            }),
-            rejectUnauthorized: false,
-            signRequest: CLIENT.signRequest
-        });
-        httpClient.keyId = CLIENT.keyId;
-        httpClient.privateKey = CLIENT.privateKey;
-        httpClient.publicKey = CLIENT.publicKey;
-
-        let MACHINE_UUID;
-
-        common.getTestImage(CLIENT, function (err, img) {
-            suite.ifError(err, 'getTestImage');
-            if (err) {
-                suite.fail('Test image not available');
-                suite.end();
-                return;
-            }
-            suite.ok(img.id, 'img.id: ' + img.id);
-            const IMAGE_UUID = img.id;
-
-            machinesCommon.createMachine(suite, CLIENT, {
-                image: IMAGE_UUID,
+    const context = {};
+    vasync.pipeline({arg: context, funcs: [
+        function setup(ctx, next) {
+            common.setup({
+                clientApiVersion: '~9'
+            }, function (_, clients, server) {
+                ctx.clients = clients;
+                ctx.client = clients.user;
+                ctx.other = clients.other;
+                ctx.server = server;
+                ctx.httpClient = makeHttpClient(ctx.client);
+                ctx.httpClientOther = makeHttpClient(ctx.other);
+                next();
+            });
+        },
+        function getImage(ctx, next) {
+            common.getTestImage(ctx.client, function (err, img) {
+                suite.ifError(err, 'getTestImage');
+                if (err) {
+                    suite.fail('Test image not available');
+                    suite.end();
+                    return;
```

Comment:
need to call `next(err);` here as well?

Also, same applies to other functions in the pipeline.

Reply:
I think the suite failing would make it all to finish. Still, I've added it.